File: Basic Challenge/2 - AddTwoNumbers/solution.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* addTwoNumbers(struct ListNode* l1, struct ListNode* l2)
{
    struct ListNode* head = l1;
    struct ListNode* before;
    int ret = 0;
    while(l1 && l2)
    {
        if((l1->val+=l2->val + ret) >= 10)
        {
            l1->val %= 10;
            ret = 1;
        }
        else
        {
            ret = 0;
        }
        before = l1;
        l1=l1->next;
        l2=l2->next;
    }
    if(l2)
    {
        before->next = l2; //rattache la suite de l2
        while(ret && l2)
        {
            if ((l2->val+=ret) >= 10)
            {
                l2->val%=10;
                ret = 1;
                before = l2;
                l2=l2->next;
            }
            else
            {
                ret =0;
            }
        }
        if(ret)
        {
            before->next = malloc(sizeof(struct ListNode));
            before->next->val = ret;
            before->next->next = NULL;
        }
    }else
    {
        while(ret && l1)
        {
            if ((l1->val+=ret) >= 10)
            {
                l1->val%=10;
                ret = 1;
                before = l1;
                l1=l1->next;
            }
            else
            {
                ret = 0;
            }
        }
        if(ret)
        {
            before->next = malloc(sizeof(struct ListNode));
            before->next->val = ret;
            before->next->next = NULL;
        }
    }
    return head;
}

File: Basic Challenge/2 - AddTwoNumbers/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        add_value = 0
        first_N = None
        prev_N = None
        while((l1 != None or l2 != None) or add_value!=0):
            if(l1 != None):
                add_value += l1.val
                l1 = l1.next
            if(l2 != None):
                add_value += l2.val
                l2 = l2.next
                
            new_N = ListNode(val=add_value%10)
            add_value = add_value//10
            if(prev_N!=None):
                prev_N.next = new_N
            else:
                first_N = new_N
            
            prev_N = new_N

        return first_N
